src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IArgument.cs

```diff
@@ -2250,6 +2250,442 @@ static void M2(int x)
             Assert.Throws<ArgumentException>(() => argument.GetOutConversion());
         }
 
+        [Fact]
+        public void DirectlyBindArgument_InvocationExpression()
+        {
+            string source = @"
+class P
+{
+    static void M1()
+    {
+        M2(/*<bind>*/1/*</bind>*/);
+    }
+    static void M2(int i) { }
+}
+";
+            string expectedOperationTree = @"
+IArgument (ArgumentKind.Explicit, Matching Parameter: i) (OperationKind.Argument) (Syntax: '1')
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
+  InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+  OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<ArgumentSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void DirectlyBindRefArgument_InvocationExpression()
+        {
+            string source = @"
+class P
+{
+    static void M1()
+    {
+        int i = 0;
+        M2(/*<bind>*/ref i/*</bind>*/);
+    }
+    static void M2(ref int i) { }
+}
+";
+            string expectedOperationTree = @"
+IArgument (ArgumentKind.Explicit, Matching Parameter: i) (OperationKind.Argument) (Syntax: 'ref i')
```

Comment:
That's pretty useful. I was just wondering about this a couple of days ago, do we have public API to show how arguments and parameters were matched...

Reply:
tagging @genlu who did argument work.